common/views/components/IIIFViewerPrototype/ViewerBottomBarPrototype.tsx

```diff
@@ -0,0 +1,178 @@
+    outline: none;
+    transition: all ${props.theme.transitionProperties};
+
+    .btn__text {
+      position: absolute;
+      right: 100%;
+      @media (min-width: ${props.theme.sizes.large}px) {
+        position: static;
+      }
+    }
+
+    &:not([disabled]):hover {
+      border: 2px solid ${props.theme.color('white')};
+    }
+  `}
+
+  ${props =>
+    !props.isDark &&
+    `
+    background: ${props.theme.color('white')};
+    color: ${props.theme.color('green')};
+    border: 1px solid ${props.theme.color('green')};
+
+    &:not([disabled]):hover,
+    &:not([disabled]):focus {
+      background: ${props.theme.color('green')};
+      color: ${props.theme.color('white')};
+    }
+  `}
+`;
+
+const BottomBar = styled.div`
+  position: relative;
+  z-index: 3;
+  background: ${props => lighten(0.14, props.theme.color('viewerBlack'))};
+  color: ${props => props.theme.color('white')};
+  display: flex;
+  justify-content: space-between;
+`;
+
+const LeftZone = styled(Space).attrs({
+  v: { size: 's', properties: ['padding-top', 'padding-bottom'] },
+})`
+  display: flex;
+  justify-content: flex-start;
+  align-items: center;
+`;
+
+const RightZone = styled(Space).attrs({
+  v: { size: 's', properties: ['padding-top', 'padding-bottom'] },
+})`
+  display: flex;
+  justify-content: flex-end;
+  align-items: center;
+`;
+
+type Props = {
+  viewToggleRef: any;
+  viewerRef: any;
+};
+
+const ViewerBottomBar: FunctionComponent<Props> = ({
+  viewToggleRef,
+  viewerRef,
+}: Props) => {
+  const { isEnhanced } = useContext(AppContext);
+  const {
+    canvases,
+    gridVisible,
+    setGridVisible,
+    work,
+    showZoomed,
+  } = useContext(ItemViewerContext);
+  return (
+    <BottomBar className="flex">
+      {isEnhanced && canvases && canvases.length > 1 && (
+        <LeftZone>
+          {!showZoomed && (
+            <Space
+              h={{ size: 's', properties: ['margin-left'] }}
+              className={classNames({
+                'flex flex--v-center flex--h-center': true,
+              })}
+            >
+              <ShameButton
+                isDark
+                ref={viewToggleRef}
+                onClick={() => {
+                  setGridVisible(!gridVisible);
+                  trackEvent({
+                    category: 'Control',
+                    action: `clicked work viewer ${
+                      gridVisible ? '"Detail view"' : '"View all"'
+                    } button`,
+                    label: `${work.id}`,
+                  });
+                }}
+              >
+                <Icon name={gridVisible ? 'detailView' : 'gridView'} />
+                <span className={`btn__text`}>
+                  {gridVisible ? 'Detail view' : 'View all'}
+                </span>
+              </ShameButton>
+            </Space>
+          )}
+        </LeftZone>
+      )}
+
+      <RightZone>
+        {isEnhanced && (
+          <div className="flex flex--v-center">
+            {document &&
```

Comment:
Worth wrapping this up somewhere?

Reply:
Like a `useHasFullscreen` hook, or similar?